src/components/App.tsx

```diff
@@ -751,6 +752,35 @@ function App() {
     if (messages) navigator.clipboard.writeText(messages);
   };
 
+  /*//////////////////////////////////////////////////////////////
+                         RENAME NODE LOGIC
+  //////////////////////////////////////////////////////////////*/
+
+  // TODO: Prevent node name to be overridden when GPT node is edited
+
+  const [nodeToRename, setNodeToRename] = useState<Node>();
+
+  const toggleShowRenameInput = () => {
+    const selectedNode = nodes.find((node) => node.selected);
+
+    setNodeToRename(selectedNode || undefined);
+  };
+
+  const renameNode = (node: Node, label: string) => {
+    takeSnapshot();
+
+    setNodes((nodes) =>
+      modifyFluxNode(nodes, {
```

Comment:
would prefer if we made a new func here for updating the label. its my bad for naming this func so generically— it really should be called `modifyFluxNodeText`